lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -148,6 +148,10 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             removeAWSChunked(request.headers['content-encoding']);
         metadataStoreParams.expires = request.headers.expires;
         metadataStoreParams.tagging = request.headers['x-amz-tagging'];
+        if (bucketMD.getObjectLockConfiguration()) {
```

Comment:
This method sounds like it returns some sort of object or structure but is being used as a conditional. I just want to make sure this is in fact what you want to do and it isn't some sort of typo.

Reply:
Yup, it's intentional although I can see why it might not look quite right, I'm gonna tweak this a bit.